### PR TITLE
Support cross region restore on azurerm_recovery_service_vault

### LINKED ISSUE #15753: Support for Cross Region Restore on RSVs
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I'd like to be able to turn on Cross Region Restore on RSVs.

### New or Affected Resource(s)

* azurerm_recovery_service_vault

### Potential Terraform Configuration

```hcl
resource "azurerm_recovery_services_vault" "vault" {
  name                = "example-recovery-vault"
  location            = azurerm_resource_group.rg.location
  resource_group_name = azurerm_resource_group.rg.name
  sku                 = "Standard"

  cross_region_restore = true
}
```

### PR DESCRIPTION
**Pull request by josh-barker**
Addresses #15753 

(fixes #15753 )

### CODE REVIEW

File: internal/services/recoveryservices/recovery_services_vault_resource.go [hunk 1]
```diff
@@ -117,6 +117,12 @@ func resourceRecoveryServicesVault() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"cross_region_restore": {
```
Comment (katbyte) [inline internal/services/recoveryservices/recovery_services_vault_resource.go:120]: could we make this consistent with other booleans in the provider?
```suggestion
			"cross_region_restore_enabled": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @josh-barker 0 one small comment on the property name I've left inline

**Comment by josh-barker** [comment]
Hey @katbyte, no worries. I've made the adjustment.

**Review by katbyte — APPROVED** [review]
Thanks @josh-barker - LGTM now! 🚀

**Pull request merged**